### DISCUSSION TITLE
EEPROM on FAT file system

### DISCUSSION THREAD

**Discussion by jnmar** [micropython/micropython#14459] [Hardware & Peripherals]
I have successfully implemented the eeprom driver from Peter Hinch [https://github.com/peterhinch/micropython_eeprom/blob/master/eeprom/i2c/I2C.md](url) on a NUCLEO STM32F767ZI2. The EEPROM chip is an ST M24512-RP on I2C. Virtual filesytem is FAT. I followed the guidelines:
```
import os
from machine import I2C
from eeprom_i2c import EEPROM, M24512RP

eep = eeprom_i2c.EEPROM(I2C(1), eeprom_i2c.T24C512, max_chips_count=1)
vfs.mount(eep, "/eeprom")
```
Files can be written and read by the MicroPython App. However, I though it would have been possible to see the virtual drive from the host (MacOs), but only the flash (PYBFLASH) is visible from the host, through the flash programming USB port. I checked documentation at: [https://docs.micropython.org/en/latest/reference/filesystem.html#filesystems](url), which says: 

> The main advantage of the FAT filesystem is that it can be accessed over USB MSC on supported boards (e.g. STM32) without any additional drivers required on the host PC.

Are MSC and the flash programming link the same ?
Any thoughts ? Thanks.

**Comment by peterhinch** [comment]
I don't know any way of doing this. My personal view is that MSC drives are the spawn of the Devil for [this reason](https://github.com/peterhinch/micropython-samples/blob/master/README.md#15-a-usb-pitfall). However it would be immensely useful to be able to mount a drive at the REPL or by a script running after boot and then access it via `mpremote`. This does not seem possible.

**Comment by andrewleech** [comment]
Yeah as @peterhinch mentions, mass storage pretty much always results in corrupted files. 

If you've written your eeprom driver as a https://docs.micropython.org/en/latest/library/vfs.html#vfs.AbstractBlockDev
Then you can use VFS to format the drive and mount it as a real filesystem internally.

You can then use `mpremote cp` or  `mpremote edit` to manage files. It'd be nice to add a "full" file browser in mpremote, but it doesn't have one yet. 

Longer term with the new dynamic usb handling it should be possible to write a protocol for MTP which could be used instead of MSC, MTP operates more at all application / file level so could give real mounted filesystems on host PC that's safe for the micropython filesystem.

**Reply by peterhinch** [reply]
The driver used by @jnmar (which I wrote) does allow mounting as a real filesystem which is fully accessible udsing MP on the device.

Please could you explain how to access this with `mpremote`. For a simple example consider the Pyboard which has filesystems on `/flash` and `/sd`. I have yet to find a way to access `/sd` from `mpremote`. `mpremote cp foo.py :` copies to `/flash`.
